### PR TITLE
Updated dough to latest

### PR DESCRIPTION
**Pull request by Sfiguz7**
## Description

PlotSquared module was updated in dough, this updates the pom to reflect that update.

## Proposed changes

Update dough to the latest version

## Related Issues (if applicable)

N/A

## Checklist

- [ ] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [ ] I have made sure that the proposed changes do not break compatibility across the supported Minecraft versions (1.16.* - 1.20.*).
- [ ] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

**Comment by J3fftw1** [comment]
This PR fixes https://github.com/Slimefun/Slimefun4/issues/4117

File: pom.xml [hunk 1]
```diff
@@ -344,7 +344,7 @@
         <dependency>
             <groupId>com.github.baked-libs.dough</groupId>
```
Comment (J3fftw1) [inline pom.xml:345]:
```suggestion
            <groupId>com.github.slimefun.dough</groupId>
```

**Review by J3fftw1 — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 2]
```diff
@@ -342,9 +342,9 @@
 
         <!-- Shaded packages -->
         <dependency>
-            <groupId>com.github.baked-libs.dough</groupId>
+            <groupId>com.github.slimefun.dough</groupId>
             <artifactId>dough-api</artifactId>
-            <version>e99d02a8ff</version>
+            <version>b25689564d</version>
```
Comment (J3fftw1) [inline pom.xml:347]:
```suggestion
            <version>6f7a534e9d</version>
```

**Review by J3fftw1 — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 3]
```diff
@@ -342,9 +342,9 @@
 
         <!-- Shaded packages -->
         <dependency>
-            <groupId>com.github.baked-libs.dough</groupId>
+            <groupId>com.github.slimefun.dough</groupId>
             <artifactId>dough-api</artifactId>
-            <version>e99d02a8ff</version>
+            <version>6f7a534e9d</version>
```
Comment (J3fftw1) [inline pom.xml:347]:
```suggestion
            <version>baf2d79f62</version>
```

**Review by WalshyDev — APPROVED** [review]

**Review by J3fftw1 — APPROVED** [review]

**Pull request merged**